src/Interpreters/Streaming/SubstituteStreamingFunction.cpp

```diff
@@ -153,19 +158,25 @@ void StreamingFunctionData::visit(DB::ASTFunction & func, DB::ASTPtr)
                     if (func.name.ends_with("_if"))
                         --delta_pos;
 
-                    /// Make _tp_delta as the last argument to avoid unused column elimination for query like below
-                    /// SELECT count(), avg(i) FROM (SELECT i, _tp_delta FROM versioned_kv) GROUP BY i; =>
-                    /// SELECT __count_retract(_tp_delta), __avg_retract(i, _tp_delta) FROM (SELECT i, _tp_delta FROM versioned_kv) GROUP BY i; =>
-                    if (func.name.starts_with("__count_retract") && delta_pos - func.arguments->children.begin() > 0)
-                        /// Fix for nullable since this substitution is not equal
-                        func.arguments->children[0] = std::make_shared<ASTIdentifier>(ProtonConsts::RESERVED_DELTA_FLAG);
-                    else
-                        func.arguments->children.insert(delta_pos, std::make_shared<ASTIdentifier>(ProtonConsts::RESERVED_DELTA_FLAG));
+                    func.arguments->children.insert(delta_pos, std::make_shared<ASTIdentifier>(ProtonConsts::RESERVED_DELTA_FLAG));
                 }
 
                 return;
             }
         }
+        else
+        {
+            /// replace `<aggr>_distinct[_combinator]` ==> `<aggr>_distinct_streaming[_combinator]` for streaming query
+            if (!boost::algorithm::contains(func.name, "_distinct_streaming")) {
+                if (boost::algorithm::contains(func.name, "_distinct")) {
+                    std::string new_name = func.name;
```

Comment:
```c++
if (func_name_lower.ends_with("_distinct"))
    substitueFunction(func, func_name_lower + "_streaming");
```

Reply:
`distinct_if` ?

Reply:
user input: distinct_streaming_if 
we shall skip the change?
so i have to check not contain streaming ?

Reply:
https://godbolt.org/z/xfh1PhKPx  no need to consider user input our internal function naming?
like `distinct_streaming_if`